docs: Add API guidance to component overview page (#1910)
cultureamp/kaizen-design-system: site/docs/components/overview.mdx

The status of these can be found on the documentation page for each component, or on the [Component health](/component-health) page, which shows the status of all components.

### API

> - **Avoid language overloading**: Avoid prop names that are used in HTML, such as `type`.

Reviewer: ```suggestion
- **Avoid language overloading**: Choose names that are self-describing that continue to be clear in slightly different contexts. For example, `type`, has a specific meaning within the context of a form, but it could also be used to describe a different appearance of a component. A good rule of thumb is to avoid names that are used in HTML.
```

> - **Lean on existing mental models**: Things like input `onChange` should not stray from the default behavior that React has defined.

Reviewer: ```suggestion
- **Lean on existing mental models**: Lean into established naming conventions where it makes sense to do so. Things like input `onChange` should not stray from the default behavior that React has defined. 
```

> - **Encapsulation:** Components should only be concerned about their own form and function and not about how they appear in relation to other components.

Reviewer: I'm curious if this is enough information for non-design system folk. Should this include an example? The best one I can think of is no padding / margin around the component.

- **Favor composition**: This allows us to have a more succinct API that encourages code reuse.

> - **RTL support**: Avoid using left/right props or values, instead use start/end and support RTL automatically.

Reviewer: ```suggestion
- **RTL support**: Avoid using left/right props or values, instead use start/end and support RTL automatically. See [CSS Logical Properties](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Logical_Properties) for more information.
```

Reviewer: Some folks might not have come across logical properties just yet :)

Author: 🤔 this isn't necessarily tied to that. Most people are familiar with start/end over left/right with flexbox, and we don't actually have to use logical properties under the hood to support start/end either / we can't because of legacy Edge.

Reviewer: Ah! My mistake - ignore this.

> - **Allowing ReactNode**: Favor allowing ReactNode values over strings where possible, for greater flexibility.

Reviewer: I know there's a few people that will be pleased about this change :D cc/ @OliBridgman ;)

#### Common props
| Name | Description | Example |
| --- | --- | --- |
| `tag` | Specifies which HTML element the component should render as. | `<Heading tag="h2">` |

> | `mood` | Dictates the tone of the component, usually affecting colors. | `<BrandMoment mood="positive">` |

Reviewer: ```suggestion
| `mood` | [Dictates the tone of the component](https://dev.cultureamp.design/api-guidelines/guidelines/moods/), usually affecting colors. | `<BrandMoment mood="positive">` |
```
